packages/common/src/core/schemas/internal/getUiSchemaProps.ts

```diff
@@ -0,0 +1,15 @@
+import { deepGetPropValues } from "../../../objects/deepGetPropValues";
+import { IUiSchema } from "../types";
+
+/**
+ * Get all the property names from a UI schema.
+ *
+ * Used to subset a json schema to only those properties used in a UI Schema
+ * @param uiSchema
+ * @returns
+ */
+export function getUiSchemaProps(uiSchema: IUiSchema): string[] {
+  return deepGetPropValues(uiSchema, "scope").map((scope: string) => {
+    return scope.split("/").pop();
```

Comment:
I think we're going to want to use the logic you'd previously written to grab the 2nd entry:

```ts
scope.split("/")[2]
```

because once we get into schema properties of type `object`, `uiSchema` scopes can look like `/properties/link/properties/href`. For example, given the following schema: 

<details>
  <summary>Schema</summary>

```ts
const schema = {
  required: ['title'],
  type: 'object',
  properties: {
    ...,
    link: {
      type: 'object',
      properties: {
        href: {
          type: 'string',
          format: 'url'
        },
        title: {
          type: 'string'
        }
      }
    }
  }
};
```
</details>

The `uiSchema` would look something like: 
<details>
  <summary>uiSchema</summary>

```ts
const uiSchema = {
  type: 'Layout',
  elements: [
    ...,
    {
      labelKey: 'link.href',
      scope: '/properties/link/properties/href',
      type: 'Control'
    },
    {
      labelKey: 'link.title',
      scope: '/properties/link/properties/title',
      type: 'Control'
    }
  ]
}
```
</details>

and from this function, I believe we'd want to return the "link" property from the scope as opposed to "title" and "href"